### PR TITLE
fix bug "aliases not recognized when federating list responses"

### LINKED ISSUE #181: Aliases not recognized when federating list responses
Federating list responses does not work as expected.

To illustrate the issue here an example from the tests: `TestExecutor_insertIntoListFragmentSpreads`
**GraphQL query**
```graphql
{
  # entry point photos is aliased here
  gallery: photos{                       # <--Query.services @ serviceA, list
    ...photosFragment
  }
}
fragment photosFragment on Photo {
  createdBy {
    firstName                            # <-- User.firstName @ serviceA
    address                              # <-- User.address @ serviceB
  }
}
```

**Expected result:**
```json
{
  "gallery": [
    {
      "createdBy": {
        "address": "address-1",
        "firstName": "John",
        "id": "1"
      }
    },
    {
      "createdBy": {
        "address": "address-2",
        "firstName": "Jane",
        "id": "2"
      }
    }
  ]
}
```

**Current result:**
```json
{
  "gallery": [
    {
      "createdBy": {
        "firstName": "John",
        "id": "1"
      }
    },
    {
      "createdBy": {
        "firstName": "Jane",
        "id": "2"
      }
    }
  ],
  "photos": [
    {
      "createdBy": {
        "address": "address-1"
      }
    },
    {
      "createdBy": {
        "address": "address-2"
      }
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by sGy1980de**
This fixes #181 

add check for alias on list elements

### CODE REVIEW

**Comment by sGy1980de** [comment]
Seems like there's a test with flaky coverage. It's complaining about 2 lines in `http.go` I did not touch at all.

File: execute_test.go [hunk 1]
```diff
@@ -513,6 +513,150 @@ func TestExecutor_insertIntoListFragmentSpreads(t *testing.T) {
...
+										},
+									},
+								},
+							},
+						},
+						Then: []*QueryPlanStep{
+							// a query to satisfy User.address
+							{
+								ParentType:     "User",
+								InsertionPoint: []string{"gallery", "createdBy"}, // photo is the query name here
```
Comment (JohnStarich) [inline execute_test.go:597]:
```suggestion
								InsertionPoint: []string{"gallery", "createdBy"}, // photo is the query name here, but gallery is the alias
```

**Review by JohnStarich — COMMENTED** [review]
Thanks @sGy1980de for the new tests and the fix! 🚀 

Do I understand correctly that the goal of `TestExecutor_insertIntoAliasedListFragmentSpreads` is to test the simpler case and the 2nd test is only used to cover a multi-step plan scenario?

**Comment by JohnStarich** [comment]
Other than my question and these really minor comment updates, this looks great 💯 

Don't worry about the coverage check, I agree it's probably a flaky coverage spot.

**Comment by sGy1980de** [comment]
Hey @JohnStarich,

to be honest, I just spotted the  `TestExecutor_insertIntoAliasedListFragmentSpreads`  first. I was scanning the tests for examples where the code would hit the problem. Later I found the other example and decided to adopt this one as well, since there this flip/flop (service A/Service B) thingy kicks in.

Thx for the review

**Review by JohnStarich — APPROVED** [review]
LGTM 👍

**Comment by JohnStarich** [comment]
Your fix is released as `v0.3.4` 🎉

**Pull request merged**